framework/include/controls/ControllableParameter.h

```diff
@@ -0,0 +1,157 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef CONTROLLABLEPARAMETER_H
+#define CONTROLLABLEPARAMETER_H
+
+// MOOSE includes
+#include "InputParameters.h"
+#include "MooseObjectName.h"
+#include "MooseObjectParameterName.h"
+
+/**
+ * A class for accessing controllable InputParameters.
+ *
+ * This class is used within Control objects for returning sets
+ * of InputParaemeters.
+ */
+template<typename T>
+class ControllableParameter
```

Comment:
Hmmmm.... I actually envisioned this completely differently.  I don't see a need for this class at all.  You shouldn't "get" a `ControllableParameter` at all.... just call a function like:

```
setControllableValue(name, value)
```

And then that function would go off and set all of the values that match name...

It's kind of weird to get an object and then have to call `set()` on it like this.  It doesn't match anything else in MOOSE... and won't translate well to a scripting language either.

But - I understand what you were going for here... and I'm willing to be wrong.